File: patterns/javascript-patterns/pattern106.js
// Pattern 106: ...

// *   *   *   *
// *   *   *   *
// * * * * * * *
// *           *
// *           *
// *           *
// * * * * * * *

let n = 7;

for (let i = 1; i <= n; i++) {
    let row = "";
    for (let j = 1; j <= n; j++) {
        if (
            (i === Math.floor(n / 2) || i === n) ||
            ((j === 1 || j === n) && i >= Math.floor(n / 2)) ||
            (j % 2 === 1 && i <= Math.floor(n / 2))
        ) {
            row += "* ";
        } else {
            row += "  ";
        }
    }
    console.log(row);
}

File: patterns/python-patterns/pattern106.py
# Pattern 106: ...

# *   *   *   *
# *   *   *   *
# * * * * * * *
# *           *
# *           *
# *           *
# * * * * * * *

n = 7

for i in range(1, n + 1):
    for j in range(1, n + 1):
        if (i == n // 2 or i == n) or \
           ((j == 1 or j == n) and i >= n // 2) or \
           (j % 2 == 1 and i <= n // 2):
            print("* ", end="")
        else:
            print("  ", end="")
    print()
